Documentation/microsoft-connector.md

```diff
@@ -0,0 +1,111 @@
+# Authentication through Microsoft
+
+## Overview
+
+One of the login options for dex uses the Microsoft OAuth2 flow to identify the
+end user through their Microsoft account.
+
+When a client redeems a refresh token through dex, dex will re-query Microsoft
+to update user information in the ID Token. To do this, __dex stores a readonly
+Microsoft access and refresh tokens in its backing datastore.__ Users that
+reject dex's access through Microsoft will also revoke all dex clients which
+authenticated them through Microsoft.
+
+### Caveats
+
+`groups` claim in dex is only supported when `tenant` is specified in Microsoft
+connector config. In order for Dex to be able to list groups on behalf of
+logged in user, an explicit organization administrator consent is required. To
+obtain the consent do the following:
+
+  - when registering dex application on [https://apps.dev.microsoft.com]() add
```

Comment:
markdown: those links don't go anywhere like this. [Random markdown cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links) suggests `https://apps.dev.microsoft.com` or `<https://apps.dev.microsoft.com>`

tests:
- https://apps.dev.microsoft.com (this is without anything)
- <https://apps.dev.microsoft.com> (this is wrapped in `<..>`)